src/schema/validate.rs

```diff
@@ -0,0 +1,1001 @@
+                            return Err(ValidationError::OutOfRange {
+                                value: int_val.to_string().into(),
+                                min: minimum.map(|m| m.to_string().into()),
+                                max: Some(max.to_string().into()),
+                                path: path.into(),
+                            });
+                        }
+                    }
+                    Ok(())
+                } else {
+                    Err(ValidationError::TypeMismatch {
+                        expected: "integer".into(),
+                        actual: "non-integer number".into(),
+                        path: path.into(),
+                    })
+                }
+            }
+            _ => Err(ValidationError::TypeMismatch {
+                expected: "integer".into(),
+                actual: Self::value_type_name(value),
+                path: path.into(),
+            }),
+        }
+    }
+
+    fn validate_number(
+        value: &Value,
+        minimum: Option<f64>,
+        maximum: Option<f64>,
+        path: &str,
+    ) -> Result<(), ValidationError> {
+        match value {
+            Value::Number(num) => {
+                if let Some(float_val) = num.as_f64() {
+                    if let Some(min) = minimum {
+                        if float_val < min {
+                            return Err(ValidationError::OutOfRange {
+                                value: float_val.to_string().into(),
+                                min: Some(min.to_string().into()),
+                                max: maximum.map(|m| m.to_string().into()),
+                                path: path.into(),
+                            });
+                        }
+                    }
+                    if let Some(max) = maximum {
+                        if float_val > max {
+                            return Err(ValidationError::OutOfRange {
+                                value: float_val.to_string().into(),
+                                min: minimum.map(|m| m.to_string().into()),
+                                max: Some(max.to_string().into()),
+                                path: path.to_string().into(),
+                            });
+                        }
+                    }
+                    Ok(())
+                } else {
+                    Err(ValidationError::TypeMismatch {
+                        expected: "number".into(),
+                        actual: "non-numeric value".into(),
+                        path: path.into(),
+                    })
+                }
+            }
+            _ => Err(ValidationError::TypeMismatch {
+                expected: "number".into(),
+                actual: Self::value_type_name(value),
+                path: path.into(),
+            }),
+        }
+    }
+
+    fn validate_boolean(value: &Value, path: &str) -> Result<(), ValidationError> {
+        match value {
```

Comment:
asking to understand: 

Why would be use the match statement here vs passing a concrete type to this method?

Reply:
Not much difference. If I had written from scratch, I would probably have written it the way you are suggesting - pass a boolean here.

The current version is more for consistency - where each validate_ method takes a Value.
But I don't think the consistency matters that much either.